odoo/fields.py

```diff
@@ -1398,10 +1398,17 @@ def _setup_related_full(self, model):
 
     def convert_to_column(self, value, record, values=None, validate=True):
         # retrieve currency from values or record
+        c_field = record._fields[self.currency_field]
         if values and self.currency_field in values:
-            field = record._fields[self.currency_field]
-            currency = field.convert_to_cache(values[self.currency_field], record, validate)
-            currency = field.convert_to_record(currency, record)
+            currency = c_field.convert_to_cache(values[self.currency_field], record, validate)
+            currency = c_field.convert_to_record(currency, record)
+        elif values and c_field.related and c_field.related[0] in values:
+            # c_field.related[0] is a relation field, let convert_to_record to return a appropriate recordset
```

Comment:
-> `# c_field.related[0] is a relational field, use convert_to_record to return an appropriated recordset`
(yes it is my own comment...)